components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2122,13 +2126,26 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
             },
         );
         if (isCreateStripeCustomerOnUsageEnabled) {
+            let customer: StripeCustomer | undefined;
+            try {
+                customer = (await this.billingService.getStripeCustomer({ attributionId })).customer;
+            } catch (e) {
+                console.error(e);
+            }
+            if (customer) {
+                // NOTE: this is a temporary workaround, as long as we're not automatically re-create the customer
+                // entity on Stripe to support a switch of currencies, we're taking an exit here.
+                if (customer.currency !== currency) {
+                    throw new ResponseError(
+                        ErrorCodes.SUBSCRIPTION_ERROR,
+                        `Cannot re-subscribe with a different currency. Please contact support.`,
```

Comment:
```suggestion
                        `Your previous subscription was in ${customer.currency} but your new subscription is in ${currency}. Please use the same currency. If you'd like to change currencies, please contact our support.`,
```

Reply:
That's a bit too long, see the rendering of messages in a narrow box. 
But I get the point.
Going to improve for more context.